sktime/datasets/base.py

```diff
@@ -31,12 +40,114 @@
 MODULE = os.path.dirname(__file__)
 
 
+def _url_to_extract(url, extract_path=None):
+    """
+    Helper function for downloading and unzipping datasets
+    This code was modified from
+    https://github.com/tslearn-team/tslearn/blob/775daddb476b4ab02268a6751da417b8f0711140/tslearn/datasets.py#L28
+    Parameters
+    ----------
+    url : string
+        Url pointing to file to download
+    extract_path : string, optional (default: None)
+        path to extract downloaded zip to, None defaults
+        to sktime/datasets/data
+
+    Returns
+    -------
+    extract_path : string or None
+        if successful, string containing the path of the extracted file, None
+        if it wasn't succesful
+
+    """
+    file_name = os.path.basename(url)
+    dl_dir = tempfile.mkdtemp()
+    zip_file_name = os.path.join(dl_dir, file_name)
+    if extract_path is None:
+        extract_path = os.path.join(MODULE, "data/%s/" % file_name.split(".")[0])
+    urlretrieve(url, zip_file_name)
+    try:
+        if not os.path.exists(extract_path):
+            os.makedirs(extract_path)
+        zipfile.ZipFile(zip_file_name, "r").extractall(extract_path)
+        shutil.rmtree(dl_dir)
+        return extract_path
+    except zipfile.BadZipFile:
+        shutil.rmtree(dl_dir)
+        raise Exception("Corrupted zip file, check URL is valid")
+        return None
+
+
+def _list_downloaded_datasets():
+    """
+    Returns a list of all the currently downloaded datasets
+    Modified version of
+    https://github.com/tslearn-team/tslearn/blob/775daddb476b4ab02268a6751da417b8f0711140/tslearn/datasets.py#L250
+
+    Returns
+    -------
+    datasets : List
+        List of the names of datasets downloaded
+
+    """
+    data_dir = os.path.join(MODULE, DIRNAME)
+    datasets = [
+        path
+        for path in os.listdir(data_dir)
+        if os.path.isdir(os.path.join(data_dir, path))
+    ]
+    return datasets
+
+
+def _list_all_datasets():
```

Comment:
I ended up removing that function and just checking to see if the exception in `_download_and_extract` was raised. I did originally want to have it check the HTTP status code for the download URL, but it appears every URL including the broken ones were returning status code 200 instead of 404s.